File: Scala/src/main/scala/parrot/NorwegianBlueParrot.scala
package parrot

class NorwegianBlueParrot(voltage: Double, isNailed: Boolean) extends Parrot {
  override def getSpeed: Double = if (isNailed) 0 else getBaseSpeed(voltage)
  private def getBaseSpeed(voltage: Double): Double = Math.min(24.0, voltage * getBaseSpeed)
}

File: Kotlin/src/main/kotlin/parrot/NorwegianBlueParrot.kt
package parrot

import kotlin.math.min

class NorwegianBlueParrot(private val voltage: Double, private val isNailed: Boolean) : Parrot {
    override val speed: Double
        get() = if (isNailed) 0.0 else computeBaseSpeed

    override val cry: String
        get() = if (voltage > 0) "Bzzzzzz" else "..."

    private val computeBaseSpeed: Double = min(24.0, voltage * baseSpeed)
}